### PR TITLE
Detailed Report Messages for Meet

### LINKED ISSUE #174: Detailed Reports message - Meet
Migrating over from old MITRE repo.

### PR DESCRIPTION
**Pull request by rgbrow1949**
Closes #174 

Added detailed report messages to meet, and refactored all correct unit tests. 

I still need to refactor the incorrect unit tests.

### CODE REVIEW

**Comment by rgbrow1949** [comment]
Unit tests are completed.

**Review by snarve — CHANGES_REQUESTED** [review]
Pulled the latest from the branch and there were merge conflicts, might need to merge the latest from main into the branch and resolve the conflicts.

**Comment by adhilto** [comment]
> Pulled the latest from the branch and there were merge conflicts, might need to merge the latest from main into the branch and resolve the conflicts.

This branch actually doesn't conflict with main. Github would let us know if there were conflicts, on this page down where it currently says, "All checks have passed."

That said, I think I know what happened to you, a similar thing happened to me. If you pulled the branch locally earlier, then try to pull again now, the current branch likely conflicts with the earlier version. The easiest way (that I know of) to resolve that is to just delete your local version and pull again. This is what I did (though it's worth recognizing again that there's probably a better way, but this did do the trick):
image

**Comment by snarve** [comment]
> > Pulled the latest from the branch and there were merge conflicts, might need to merge the latest from main into the branch and resolve the conflicts.
> 
> This branch actually doesn't conflict with main. Github would let us know if there were conflicts, on this page down where it currently says, "All checks have passed."
> 
> That said, I think I know what happened to you, a similar thing happened to me. If you pulled the branch locally earlier, then try to pull again now, the current branch likely conflicts with the earlier version. The easiest way (that I know of) to resolve that is to just delete your local version and pull again. This is what I did (though it's worth recognizing again that there's probably a better way, but this did do the trick): image

yeah I tried resolving the conflicts but didn't work. yes it did conflict with the earlier version of the branch. was saving deleting the branch as the last resort. will delete the branch and test. Thanks for the update.

**Comment by snarve** [comment]
> > > Pulled the latest from the branch and there were merge conflicts, might need to merge the latest from main into the branch and resolve the conflicts.
> > 
> > 
> > This branch actually doesn't conflict with main. Github would let us know if there were conflicts, on this page down where it currently says, "All checks have passed."
> > That said, I think I know what happened to you, a similar thing happened to me. If you pulled the branch locally earlier, then try to pull again now, the current branch likely conflicts with the earlier version. The easiest way (that I know of) to resolve that is to just delete your local version and pull again. This is what I did (though it's worth recognizing again that there's probably a better way, but this did do the trick): image
> 
> yeah I tried resolving the conflicts but didn't work. yes it did conflict with the earlier version of the branch. was saving deleting the branch as the last resort. will delete the branch and test. Thanks for the update.

cool, it worked out

**Review by snarve — APPROVED** [review]

File: rego/Meet.rego [hunk 1]
```diff
@@ -12,7 +12,22 @@ LogEvents := utils.GetEvents("meet_logs")
 #
 # Baseline GWS.MEET.1.1v0.1
 #--
-NonCompliantOUs1_1 contains OU if {
+UserFriendlyValues1_1 := {
+    "ALL": "all users (including users not signed in with a Google account)"
+}
+
```
Comment (adhilto) [inline rego/Meet.rego:18]: Recommend deleting stale code that isn't used anywhere
 ```suggestion
```

File: rego/Meet.rego [hunk 2]
```diff
@@ -59,7 +74,22 @@ if {
 #
 # Baseline GWS.MEET.2.1v0.1
 #--
-NonCompliantOUs2_1 contains OU if {
+UserFriendlyValues2_1 := {
+    "ALL": "any meetings, including meetings created with personal accounts"
+}
+
```
Comment (adhilto) [inline rego/Meet.rego:80]: Stale code
```suggestion
```

File: rego/Meet.rego [hunk 3]
```diff
@@ -105,7 +135,22 @@ if {
 #
 # Baseline GWS.MEET.3.1v0.1
 #--
-NonCompliantOUs3_1 contains OU if {
+UserFriendlyValues3_1 := {
+    "false": "off"
+}
+
```
Comment (adhilto) [inline rego/Meet.rego:141]:
```suggestion
```

File: rego/Meet.rego [hunk 4]
```diff
@@ -150,7 +195,22 @@ if {
 #
 # Baseline GWS.MEET.4.1v0.1
 #--
-NonCompliantOUs4_1 contains OU if {
+UserFriendlyValues4_1 := {
+    "false": "no warning label"
+}
+
```
Comment (adhilto) [inline rego/Meet.rego:201]:
```suggestion
```

File: rego/Meet.rego [hunk 5]
```diff
@@ -38,7 +53,7 @@ if {
 tests contains {
     "PolicyId": "GWS.MEET.1.1v0.1",
     "Criticality": "Shall",
-    "ReportDetails": utils.ReportDetailsOUs(NonCompliantOUs1_1),
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_1, []),
```
Comment (adhilto) [inline rego/Meet.rego:56]: Recommend adding a comment like this at least for the first setting, just so it's easier to understand what's going on.
```suggestion
    # Empty list in next line for non-compliant groups, as Meet settings can't be changed at the group level
    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_1, []),
```

**Review by adhilto — CHANGES_REQUESTED** [review]
Functionally works exactly as expected :100:. Just a few style things I'd recommend.

**Review by adhilto — APPROVED** [review]

**Pull request merged**